cdk/cmr-stac/lib/cmr-stac-stack.ts

```diff
@@ -0,0 +1,260 @@
+        SUBNET_ID_A!,
+        SUBNET_ID_B!,
+        SUBNET_ID_C!
+      ],
+      vpcId: VPC_ID!,
+    })
+
+    const lambdaSecurityGroup = SecurityGroup.fromSecurityGroupId(this, 'CmrStacSecurityGroup', CMR_SERVICE_SECURITY_GROUP_ID);
+    
+    // Environment variables to pass to application
+    const environment = {
+      CMR_URL,
+      CMR_LB_URL,
+      GRAPHQL_URL,
+      STAC_VERSION,
+      PAGE_SIZE,
+      LOG_LEVEL
+    }
+    // TODO can I make this lambda using the library
+    const stacLambdaFunction = new lambdaNodeJs.NodejsFunction(this, `StacLambdaFunction`, {
+      bundling,
+      architecture: lambda.Architecture.X86_64,
+      currentVersionOptions: {
+        removalPolicy: cdk.RemovalPolicy.RETAIN
+      },
+      entry: '../../src/handler.ts',
+      environment,
+      functionName: `cmr-stac-api-eudoro-${STAGE_NAME}-stac${logGroupSuffix}`,
+      handler: 'default',
+      logGroup: stacLogGroup,
+      role: iam.Role.fromRoleArn(this, 'CmrStacLambdaRole', cmrStacRole.attrArn),
+      memorySize,
+      runtime,
+      vpc,
+      securityGroups:[lambdaSecurityGroup],
+      timeout: cdk.Duration.seconds(30)
+    })
+
+    new logs.CfnSubscriptionFilter(this, 'StacSubscriptionFilter', {
+      destinationArn: LOG_DESTINATION_ARN,
+      filterPattern: '',
+      logGroupName: stacLogGroup.logGroupName,
+    })
+
+    const stacLambdaPermissionRegisterTarget = new lambda.CfnPermission(this, 'StacLambdaPermissionRegisterTarget', {
+      functionName: stacLambdaFunction.functionName,
+      action: 'lambda:InvokeFunction',
+      principal: 'elasticloadbalancing.amazonaws.com',
+    })
+
+    const stacAlbTargetGroup = new elasticloadbalancingv2.CfnTargetGroup(this, 'StacAlbTargetGroup', {
+      targetType: 'lambda',
+      targets: [
+        {
+          id: stacLambdaFunction.functionArn,
+        },
+      ],
+      tags: [
+        {
+          key: 'Name',
+          value: `cmr-stac-api-stac-${STAGE_NAME}-eudoro`,
+        },
+      ],
+      targetGroupAttributes: [
+        {
+          key: 'lambda.multi_value_headers.enabled',
+          value: 'false',
+        },
+      ],
+      healthCheckEnabled: false,
+    })
+
+    stacAlbTargetGroup.addDependency(stacLambdaPermissionRegisterTarget);
+
+    // Add Alb Listener rules behind the CMR load balancer
+    new elasticloadbalancingv2.CfnListenerRule(this, 'StacAlbListenerRule500', {
+      actions: [
+        {
+          type: 'forward',
+          targetGroupArn: stacAlbTargetGroup.ref,
+        },
+      ],
+      conditions: [
+        {
+          field: 'path-pattern',
+          values: [
+            '/eudorostac*',
+          ],
+        },
+        {
+          field: 'http-request-method',
+          httpRequestMethodConfig: {
+            values: [
+              'GET',
+              'POST',
+              'OPTIONS',
+            ],
+          },
+        },
+      ],
+      listenerArn: LISTENER_ARN,
+      priority: 680,
```

Comment:
One thing to note is that the deployment will fail if we keep the `alb` rule at the same priority so the first deployment (Where we need to have the `_cdk` suffix) will also need to have new rules (same ones essentially with a different priority) then in the subsequent deployment without the suffix the priority can be put back down to `500` and `501` respectively